File: image-crud/src/main/java/org/judexmars/imagecrud/dto/kafka/ImageStatusMessage.java
package org.judexmars.imagecrud.dto.kafka;

import com.fasterxml.jackson.annotation.JsonInclude;
import java.util.List;
import java.util.UUID;
import org.judexmars.imagecrud.dto.imagefilters.FilterType;

/**
 * Message that is sent to kafka after user requests applying filters.
 *
 * @param imageId   id of the image
 * @param requestId id of the user's request
 * @param filters   list of filter types
 */
public record ImageStatusMessage(
    UUID imageId,
    UUID requestId,
    @JsonInclude(JsonInclude.Include.NON_EMPTY)
    List<FilterType> filters
) {
}

File: image-processor/src/main/kotlin/org/judexmars/imageprocessor/dto/ImageStatusMessage.kt
package org.judexmars.imageprocessor.dto

import java.util.UUID

/**
 * Message that is sent to kafka after user requests applying filters.
 *
 * @param imageId   id of the image
 * @param requestId id of the user's request
 * @param filters   list of filter types
 */
data class ImageStatusMessage(
    val imageId: UUID,
    val requestId: UUID,
    val filters: List<String>,
    val props: Map<String, String>,
)
